### ISSUE TITLE
is there a button to select reinstall grub?

### ISSUE THREAD

**Issue by candrapersada** [rescuezilla/rescuezilla#225]
is there a select and unselect button to reinstall grub during restore and clone?

**Comment by shasheene** [comment]
Since v2.2, Rescuezilla follows the _default_ Clonezilla behavior of searching the partitions being restored for an existing GRUB bootloader and (if found) re-installing it.

On systems _without_ a GRUB bootloader, this operation has no effect. And on systems _with_ a GRUB bootloader, re-installing the GRUB bootloader is harmless. In fact Linux distributions themselves do this when for example the Linux kernel has been upgraded.

There is not currently any option to disable this behavior. It's the default behavior of Clonezilla's 'Beginner Mode' that has been tested by tens of million of people for two decades.

**Comment by Zlika** [comment]
This behavior may be "harmless" but it makes the restore/clone process not reproducible, the hash of some files being changed in the process. It's a showstopper for my usecase, so I'll need to find another backup tool.

**Comment by shasheene** [comment]
That's good feedback.

I would love for Rescuezilla' to have an option to restore and clone process to be able to produce a bit-for-bit identical match to the source image.

The normal operation can't do this for a number of other reasons:

1. Expands partition table to fit target disk (when target disk is larger than source)
2. Expands final partition (and grows filesystem) to fit final partition
3. Restores partition table using `sfdisk` ("scriptable fdisk") which reads a text file backup of the most important pieces of information which is reliable but is almost certainly won't reproduce any uninitialized bit patterns

...and probably many more.

A direct `dd` backup (or as I prefer for disks, `ddrescue`) is a powerful (and dangerous) command-line tool to achieve this kind of forensic imaging. You may find there's a ddrescue GUI somewhere.

Rescuezilla will eventually get ddrescue functionality (#143), but I'm not actively working on that.

**Comment by Zlika** [comment]
Hi!
I don't really need bit-for-bit clone, I just want to be able to check the hash of the files of the copy compared to the original version. And this grub update changes some files. I didn't have this problem previously when I used redo 1.x.

**Comment by Zlika** [comment]
Hi @shasheene,
Would you consider a PR with an opt-out checkbox during the restore process to not reinstall the bootloader nor update the initramfs?
I'm currently testing adding a "Reinstall Grub/Syslinux bootloader and update initramfs (if any)" checkbox at step 4 of the restore process (just below the "Overwrite partition table" checkbox). This checkbox would be checked by default, but the user could uncheck it to skip the "bootloader reinstall + initramfs update" phase.

**Comment by shasheene** [comment]
Yep will take a look (I have taken a week off work starting this weekend to close out the existing PR's and hopefully finally release a new Rescuezilla version).

I am still not fully comfortable with an "advance options" page. The option should be moved to such a page in a long term though.

By the way @Zlika have you tried Redo Rescue? It's developed by the Redo 1.x developer and doesn't do Rescuezilla's advanced reinstall of GRUB as far as I know. It may be a more suitable immediate tool for you.

**Comment by Zlika** [comment]
Redo Rescue seems to be a dead project.
An "advanced options" page would be a good idea. Maybe not as an extra "page" in the process (that would make the process more complex for beginners looking for a simple backup tool). What about a small "Advanced options" button in an upper corner of the last page, so beginners can easily ignore it, and advanced users can tweak the process to their needs?
Currently I put the "Reinstall Grub/Syslinux bootloader and update initramfs (if any)" checkbox at step 4 of the restore process, just below the "Overwrite partition table" checkbox. Even this "Overwrite partition table" may be considered as an "advanced" option (that's why there are lots of lines of text to explain the option and warn the user).
